### ISSUE TITLE
Errors when trying to import/require cytoscape with webpack.

### ISSUE THREAD

**Issue by 01AutoMonkey** [cytoscape/cytoscape.js#1091]
When trying to "require" cytoscape `var cytoscape = require('cytoscape');` with webpack (which I think is the only relevant component here, though more specifically I'm using [this fork](https://github.com/01AutoMonkey/webapp-init) of [this](https://github.com/petehunt/webpack-howto) webapp boilerplate) I get:

```console
monk01@HAL:~/dev/experi/graphs/client$ webpack --watch
[BABEL] Note: The code generator has deoptimised the styling of "/home/monk01/dev/experi/graphs/client/node_modules/cytoscape/dist/cytoscape.js" as it exceeds the max of "100KB".
[BABEL] Note: The code generator has deoptimised the styling of "/home/monk01/dev/experi/graphs/client/node_modules/cytoscape/dist/cytoscape.min.js" as it exceeds the max of "100KB".
Hash: 2f0bd9ddb31d838eabf3
Version: webpack 1.12.2
Time: 49599ms
    Asset     Size  Chunks             Chunk Names
bundle.js  1.63 MB       0  [emitted]  main
    + 212 hidden modules

WARNING in ./~/cytoscape/dist/cytoscape.js
Critical dependencies:
65:274-281 require function is used in a way in which dependencies cannot be statically extracted
67:249-256 require function is used in a way in which dependencies cannot be statically extracted
68:0-7 require function is used in a way in which dependencies cannot be statically extracted
 @ ./~/cytoscape/dist/cytoscape.js 65:274-281 67:249-256 68:0-7

WARNING in ./~/cytoscape/dist/cytoscape.js
Critical dependencies:
66:10-23 the request of a dependency is an expression
 @ ./~/cytoscape/dist/cytoscape.js 66:10-23

WARNING in ./~/cytoscape/dist/cytoscape.min.js
Critical dependencies:
16:6978-6985 require function is used in a way in which dependencies cannot be statically extracted
16:7138-7145 require function is used in a way in which dependencies cannot be statically extracted
16:7149-7156 require function is used in a way in which dependencies cannot be statically extracted
 @ ./~/cytoscape/dist/cytoscape.min.js 16:6978-6985 16:7138-7145 16:7149-7156

WARNING in ./~/cytoscape/dist/cytoscape.min.js
Critical dependencies:
16:6994-7004 the request of a dependency is an expression
 @ ./~/cytoscape/dist/cytoscape.min.js 16:6994-7004
```

And in the browser (Chrome) I get `Uncaught TypeError: Cannot set property 'cytoscape' of undefined` and in Firefox I get `TypeError: undefined has no properties`.

So looking at the `webpack --watch` output I see I'm getting a "Critical dependencies" warning, mainly:

- `require function is used in a way in which dependencies cannot be statically extracted`
- `the request of a dependency is an expression`

The lines in question seem to be:

```js
  if( typeof module !== 'undefined' && module.exports ){ // expose as a commonjs module
    module.exports = cytoscape;
  }

  if( typeof define !== 'undefined' && define.amd ){ // expose as an amd/requirejs module
    define('cytoscape', function(){
      return cytoscape;
    });
  }
```

So in short, I'm getting errors both in webpack console output and in the browser, resulting in my browser code stopping at `var cytoscape = require('cytoscape');`. Any ideas what this could be?

**Comment by 01AutoMonkey** [comment]
After some tinkering I found out that this is not really `webpack` specific but `babel-loader` specific, if I switch `babel-loader` with `jsx-loader` the code works, but I still get some non-braking errors in the webpack output. So not really a solution since jsx-loader is deprecated and one should really use babel-loader, but it will work for the moment.

So from this:

```js
module.exports = {
  entry: './modules/main.js',
  output: {
    filename: 'bundle.js'
  },
  module: {
    loaders: [
      { test: /\.js$/, loader: 'babel-loader' },
      { test: /\.scss$/, loader: "style!css!sass?sourceMap"},
      { test: /\.less$/, loader: 'style-loader!css-loader!less-loader' }, // use ! to chain loaders
      { test: /\.css$/, loader: 'style-loader!css-loader' },
      { test: /\.(png|jpg)$/, loader: 'url-loader?limit=8192' } // inline base64 URLs for <=8k images, direct URLs for the rest
    ]
  }
};
```

To this:

```js
module.exports = {
  entry: './modules/main.js',
  output: {
    filename: 'bundle.js'
  },
  module: {
    loaders: [
      { test: /\.js$/, loader: 'jsx-loader?harmony' },
      { test: /\.scss$/, loader: "style!css!sass?sourceMap"},
      { test: /\.less$/, loader: 'style-loader!css-loader!less-loader' }, // use ! to chain loaders
      { test: /\.css$/, loader: 'style-loader!css-loader' },
      { test: /\.(png|jpg)$/, loader: 'url-loader?limit=8192' } // inline base64 URLs for <=8k images, direct URLs for the rest
    ]
  }
};
```

Webpack output:

```console
monk01@HAL:~/dev/experi/graphs/client$ webpack --watch
Hash: 6048fcff5a82f34d26a1
Version: webpack 1.12.2
Time: 15997ms
    Asset     Size  Chunks             Chunk Names
bundle.js  1.81 MB       0  [emitted]  main
    + 212 hidden modules

WARNING in ./~/cytoscape/dist/cytoscape.js
Critical dependencies:
563:63-70 require function is used in a way in which dependencies cannot be statically extracted
579:59-66 require function is used in a way in which dependencies cannot be statically extracted
580:10-17 require function is used in a way in which dependencies cannot be statically extracted
 @ ./~/cytoscape/dist/cytoscape.js 563:63-70 579:59-66 580:10-17

WARNING in ./~/cytoscape/dist/cytoscape.js
Critical dependencies:
565:18-33 the request of a dependency is an expression
 @ ./~/cytoscape/dist/cytoscape.js 565:18-33

WARNING in ./~/cytoscape/dist/cytoscape.min.js
Critical dependencies:
18:1160-1167 require function is used in a way in which dependencies cannot be statically extracted
18:1296-1303 require function is used in a way in which dependencies cannot be statically extracted
18:1305-1312 require function is used in a way in which dependencies cannot be statically extracted
 @ ./~/cytoscape/dist/cytoscape.min.js 18:1160-1167 18:1296-1303 18:1305-1312

WARNING in ./~/cytoscape/dist/cytoscape.min.js
Critical dependencies:
18:1174-1184 the request of a dependency is an expression
 @ ./~/cytoscape/dist/cytoscape.min.js 18:1174-1184
```

**Comment by maxkfranz** [comment]
Cytoscape works fine with webpack.  It looks like you've configured babel-loader to go through your dependencies/node_modules.  You shouldn't be processing dependencies with babel whether it's with gulp/webpack/grunt, right?

See `exclude` https://github.com/babel/babel-loader#options
